IanGRap/Squish
# Background Eyes Test

Contributors: Coulter, Athene

Goal: Create an effect where glowing eyes can be seen opening and closing in the background of the new arena a la old scooby-doo-era cartoons.

Important Files:
- Background Eyes 1.prefab: Holds eye sprite, plus two masks that go up and down to cover it. Also has a 2D collider which is used to detect when a new potential spawn position overlaps with another eye that has already spawned.
- EyeScript.cs: Sets the eye sprite randomly from a list, controls the movement of the masks. Several variables are available to tweak so that the masks act as much like eyelashes as possible. If other differently-shaped eye sprites are introduced, they could be put in separate prefabs where the sprite mask movement values can be tweaked to fit the eye shape.
- Eye Zone.prefab: Empty game object that spawns eyes using randomized positions and spawn time intervals.
- EyeSpawner.cs: Script that controls when and where to spawn eyes.

How to test:
- Either pick a game mode scene or make a new empty one.
- Drag the Eye Zone prefab into the scene. If you put the eyes in an existing scene, it could be a good way to judge it while playing. Also, if you put the zone in an existing scene, make sure to put it a little behind everything on the z axis so that it doesn't overlap with blocks and players.
- Play the scene.
- Observe the eyes spawning, opening, closing, and being destroyed. Ideally, no eyes should be overlapping each other. Do you like how often they spawn, how quickly they open and close, etc?
